operator/pkg/helmreconciler/apply.go

```diff
@@ -35,21 +35,33 @@ import (
 
 const fieldOwnerOperator = "istio-operator"
 
+// AppliedResult is the result of applying a Manifest.
+type AppliedResult struct {
+	// processedObjects is the list of objects that were processed in this apply operation.
+	processedObjects object.K8sObjects
+	// deployed is the number of objects have been deployed which means
+	// it's in the cache and it's not changed from the cache.
+	deployed int
+}
+
+// Succeed returns true if the apply operation succeeded.
+func (r AppliedResult) Succeed() bool {
+	return len(r.processedObjects) > 0 && r.deployed > 0
```

Comment:
I think this has changed the original condition:
```
len(processedObjs) != 0 || deployedObjects > 0
```
which uses `||` instead of `&&`?